x/evm/keeper/chainKeeper.go

```diff
@@ -883,6 +875,32 @@ func (k chainKeeper) SetConfirmedEvent(ctx sdk.Context, event types.Event) error
 	return nil
 }
 
+// EnqueueConfirmedEvent enqueues the confirmed event
+func (k chainKeeper) EnqueueConfirmedEvent(ctx sdk.Context, id types.EventID) error {
+	event, ok := k.GetEvent(ctx, id)
+	if !ok {
+		return fmt.Errorf("event %s does not exist", id)
+	}
+	if event.Status != types.EventConfirmed {
+		return fmt.Errorf("event %s is not confirmed", id)
+	}
+
+	switch event.GetEvent().(type) {
+	// we no longer allow Event_ContractCall to be enqueued in the EVM module, but
+	// to be enqueued in the nexus module as a general message instead
```

Comment:
this comment is a bit confusing to me, are you saying we don't have to care about Event_ContractCall because it's treated as a general message already?

Reply:
I'm saying that `Event_ContractCall` is not allowed to be enqueued here anymore. They should be handled as nexus general message instead.